sdkproject/Assets/Mapbox/Unity/Map/AbstractMap.cs

```diff
@@ -661,6 +662,173 @@ public void SetLoadingTexture(Texture2D loadingTexture)
+			{
+				findByType = LocationPrefabFindBy.MapboxCategory,
+				categories = categories,
+				density = density,
+				prefabItemName = locationItemName,
+				spawnPrefabOptions = new SpawnPrefabOptions()
+				{
+					prefab = prefab,
+					scaleDownWithWorld = scaleDownWithWorld
+				}
+			};
+			if(_vectorData==null)
+			{
+				_vectorData = new VectorLayer();
+			}
+			if (_vectorData.LocationPrefabsLayerProperties.locationPrefabList == null)
+			{
+				_vectorData.LocationPrefabsLayerProperties.locationPrefabList = new List<PrefabItemOptions>();
+			}
+
+			if (callback != null)
+				item.OnAllPrefabsInstantiated += callback;
+			
+			if (!_vectorData.LayerProperty.vectorSubLayers.Contains(item))
+			{
+				_vectorData.LocationPrefabsLayerProperties.locationPrefabList.Add(item);
+				_vectorData.AddVectorLayer(item);
+			}
+		}
+
+		/// <summary>
+		/// Places the prefab at POI locations if its name contains the supplied string
+		/// <param name="prefab">GameObject Prefab</param>
+		/// <param name="nameString">This is the string that will be checked against the POI name to see if is contained in it, and ony those POIs will be spawned</param>
+		/// <param name="density">Density (Integer value between 1 and 30)</param>
+		/// <param name="locationItemName">Name of this location prefab item for future reference</param>
+		/// <param name="scaleDownWithWorld">Should the prefab scale up/down along with the map game object?</param>
+		/// </summary>
+		public void SpawnPrefabByName(GameObject prefab, string nameString, int density = 30, Action<List<GameObject>> callback = null, bool scaleDownWithWorld = true, string locationItemName = "New Location")
+		{
+			if (_vectorData.LayerProperty.sourceType != VectorSourceType.None || _vectorData.LayerProperty.sourceOptions.Id.Contains(MapboxDefaultVector.GetParameters(VectorSourceType.MapboxStreets).Id))
+			{
+				Debug.LogError("In order to place location prefabs please add \"mapbox.mapbox-streets-v7\" to the list of vector data sources");
+				return;
+			}
+			PrefabItemOptions item = new PrefabItemOptions()
+			{
+				findByType = LocationPrefabFindBy.POIName,
+				nameString = nameString, 
+				density = density,
+				prefabItemName = locationItemName,
+				spawnPrefabOptions = new SpawnPrefabOptions()
+				{
+					prefab = prefab,
+					scaleDownWithWorld = scaleDownWithWorld
+				}
+			};
+			if (_vectorData == null)
+			{
+				_vectorData = new VectorLayer();
+			}
+			if (_vectorData.LocationPrefabsLayerProperties.locationPrefabList == null)
+			{
+				_vectorData.LocationPrefabsLayerProperties.locationPrefabList = new List<PrefabItemOptions>();
+			}
+
+			if (callback != null)
+				item.OnAllPrefabsInstantiated += callback;
+			
+			if (!_vectorData.LayerProperty.vectorSubLayers.Contains(item))
+			{
+				_vectorData.LocationPrefabsLayerProperties.locationPrefabList.Add(item);
+				_vectorData.AddVectorLayer(item);
+			}
+		}
+
+  		#endregion
```

Comment:
Can we refactor the redundant code in all 4 prefab functions ? Only difference I see is setting up the prefabItem options.